flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -91,7 +91,7 @@ public static Builder forRow(DataStream<Row> input, TableSchema tableSchema) {
     DataType[] fieldDataTypes = tableSchema.getFieldDataTypes();
 
     DataFormatConverters.RowConverter rowConverter = new DataFormatConverters.RowConverter(fieldDataTypes);
-    return builderFor(input, rowConverter::toInternal, RowDataTypeInfo.of(rowType))
+    return builderFor(input, rowConverter::toInternal, InternalTypeInfo.of(rowType))
```

Comment:
I checked the flink codebase,  all the `RowData`'s TypeSerializer related classes are not expose to be `public`, they are marked as `Internal`.  The only way I can think of is:  Use a compactbility utility to load `InternalTypeInfo`  or `RowDataTypeInfo` class if there's exist one.

Reply:
I think this is the shim layer approach that others suggested.

Reply:
Em, It's true.